payout.go

```diff
@@ -0,0 +1,156 @@
+// PayoutListParams is the set of parameters that can be used when listing payouts.
+// For more details see https://stripe.com/docs/api#list_payouts.
+type PayoutListParams struct {
+	ListParams
+	ArrivalDate int64
+	Created     int64
+	Destination string
+	Status      PayoutStatus
+}
+
+// Payout is the resource representing a Stripe payout.
+// For more details see https://stripe.com/docs/api#payouts.
+type Payout struct {
+	ID                        string            `json:"id"`
+	Live                      bool              `json:"livemode"`
+	Amount                    int64             `json:"amount"`
+	ArrivalDate               int64             `json:"arrival_date"`
+	BalanceTransaction        *Transaction      `json:"balance_transaction"`
+	Bank                      *BankAccount      `json:"bank_account"`
+	Card                      *Card             `json:"card"`
+	Created                   int64             `json:"created"`
+	Currency                  Currency          `json:"currency"`
+	Destination               PayoutDestination `json:"destination"`
+	FailureBalanceTransaction *Transaction      `json:"failure_balance_transaction"`
+	FailCode                  PayoutFailureCode `json:"failure_code"`
+	FailMessage               string            `json:"failure_message"`
+	Meta                      map[string]string `json:"metadata"`
+	Method                    PayoutMethodType  `json:"method"`
+	SourceType                PayoutSourceType  `json:"source_type"`
+	StatementDescriptor       string            `json:"statement_descriptor"`
+	Status                    PayoutStatus      `json:"status"`
+	Type                      PayoutType        `json:"type"`
+}
+
+// PayoutList is a list of payouts as retrieved from a list endpoint.
+type PayoutList struct {
+	ListMeta
+	Values []*Payout `json:"data"`
+}
+
+// UnmarshalJSON handles deserialization of a Payout.
+// This custom unmarshaling is needed because the resulting
+// property may be an id or the full struct if it was expanded.
+func (t *Payout) UnmarshalJSON(data []byte) error {
+	type payout Payout
+	var tb payout
+	err := json.Unmarshal(data, &tb)
+	if err == nil {
+		*t = Payout(tb)
+	} else {
+		// the id is surrounded by "\" characters, so strip them
+		t.ID = string(data[1 : len(data)-1])
+	}
+
+	return nil
+}
+
+// UnmarshalJSON handles deserialization of a PayoutDestination.
+// This custom unmarshaling is needed because the specific
+// type of destination it refers to is specified in the JSON
+func (d *PayoutDestination) UnmarshalJSON(data []byte) error {
+	type dest PayoutDestination
+	var dd dest
+	err := json.Unmarshal(data, &dd)
+	if err == nil {
+		*d = PayoutDestination(dd)
+
+		switch d.Type {
+		case PayoutDestinationBankAccount:
+			json.Unmarshal(data, &d.BankAccount)
+		case PayoutDestinationCard:
+			json.Unmarshal(data, &d.Card)
+		}
+	} else {
+		// the id is surrounded by "\" characters, so strip them
+		d.ID = string(data[1 : len(data)-1])
+	}
+
+	return nil
```

Comment:
Why don't we return `err` here or handle error in `json.Unmarshal` in line 138 or 140 ?

Reply:
Good question. So you may notice that a few lines above, `data` is already run through another `json.Unmarshal`:

```
	err := json.Unmarshal(data, &dd)
```

Given that it's already been validated there (i.e. it's likely to be valid JSON, etc.), it's fairly unlikely that those other `Unmarshal` invocations are going to throw an error, and I think that's why an error check on them was originally excluded (it turns out that this is/was a common convention in the library and exists throughout a number of files).

That said, even though it's unlikely to be needed it's still good practice to include an error check anyway. I've added a patch in #509 that puts error checks in all these places.